### PR TITLE
libccd 2.0 (migrated from homebrew-science)

### PR DESCRIPTION
**Pull request by jslee02**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Migrating because of Homebrew/homebrew-science#6331

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/libccd.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  bottle do
+    cellar :any
+    sha256 "042f76f2faba70f4afd7772865014acf1e18b4596e55cbca4be4414bfdd7d7c5" => :sierra
+    sha256 "73dc3083059a205688fc3edf3c49a5277fdb3566b0d0a2e3217d43b79b14093c" => :el_capitan
+    sha256 "52cc90f9245c514ab30cf1b5d5af292dfd014c804556b0fd0f07afe431c7bf87" => :yosemite
+    sha256 "fae18bcb83cdca783aecb9084fea2803cf60a79a8c46a409c2ee6b94505dfee4" => :x86_64_linux
+  end
+
+  option "with-single-precision", "Use single precision"
```
Comment (fxcoudert) [inline Formula/libccd.rb]: Analytics show this option is never used, remove it.

File: Formula/libccd.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  depends_on "cmake" => :build
+
+  def install
+    args = std_cmake_args
+
+    if build.with? "single-precision"
+      args << "-DCCD_SINGLE=True"
+    else
+      args << "-DCCD_DOUBLE=True"
```
Comment (fxcoudert) [inline Formula/libccd.rb]: Double is the default, so this whole block can go (_“libccd is by default compiled in double precision of floating point numbers”_)

Comment (jslee02) [reply Formula/libccd.rb]: `ENABLE_DOUBLE_PRECISION=ON` option added which is [necessary for building using CMake](https://github.com/danfis/libccd/blob/64f02f741ac94fccd0fb660a5bffcbe6d01d9939/README.md#3-using-cmake):
> To compile using double precision, set the ENABLE_DOUBLE_PRECISION option:

**Comment by jslee02** [comment]
Sorry for the late response. My notification setting was unintentionally disabled for a few days. I believe feedbacks of @fxcoudert are addressed by 8b1a49e.

File: Formula/libccd.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 "fae18bcb83cdca783aecb9084fea2803cf60a79a8c46a409c2ee6b94505dfee4" => :x86_64_linux
+  end
+
+  depends_on "cmake" => :build
+
+  def install
+    args = std_cmake_args
+    args << "-DENABLE_DOUBLE_PRECISION=ON"
+
+    system "cmake", ".", *args
```
Comment (fxcoudert) [inline Formula/libccd.rb]: Would be slightly better without creating a variable:

```
system "cmake", ".", *std_cmake_args, "-DENABLE_DOUBLE_PRECISION=ON"
```

(with wrapping if it doesn't fit in 80 chars)

Comment (jslee02) [reply Formula/libccd.rb]: Done!

File: Formula/libccd.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Libccd < Formula
+  desc "Library for collision detection between two convex shapes"
+  homepage "http://libccd.danfis.cz"
```
Comment (ilovezfs) [inline Formula/libccd.rb]: trailing slash

File: Formula/libccd.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  url "https://github.com/danfis/libccd/archive/v2.0.tar.gz"
+  sha256 "1b4997e361c79262cf1fe5e1a3bf0789c9447d60b8ae2c1f945693ad574f9471"
+  revision 2
+  head "https://github.com/danfis/libccd.git"
```
Comment (ilovezfs) [inline Formula/libccd.rb]: remove head please

File: Formula/libccd.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    sha256 "042f76f2faba70f4afd7772865014acf1e18b4596e55cbca4be4414bfdd7d7c5" => :sierra
+    sha256 "73dc3083059a205688fc3edf3c49a5277fdb3566b0d0a2e3217d43b79b14093c" => :el_capitan
+    sha256 "52cc90f9245c514ab30cf1b5d5af292dfd014c804556b0fd0f07afe431c7bf87" => :yosemite
+    sha256 "fae18bcb83cdca783aecb9084fea2803cf60a79a8c46a409c2ee6b94505dfee4" => :x86_64_linux
+  end
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", ".", *std_cmake_args, "-DENABLE_DOUBLE_PRECISION=ON"
```
Comment (ilovezfs) [inline Formula/libccd.rb]:
```
system "cmake", ".", "-DENABLE_DOUBLE_PRECISION=ON", *std_cmake_args
```

File: Formula/libccd.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
+class Libccd < Formula
+  desc "Library for collision detection between two convex shapes"
```
Comment (ilovezfs) [inline Formula/libccd.rb]: I think we know it's a library based on the name of the formula

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Review by ilovezfs — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @jslee02 for your contribution to Homebrew! 🎉

**Pull request closed without merging**